community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -507,13 +506,19 @@
             "Can be used for specifying the threshold to prune logical logs after. For example \"10 days\" will " +
             "prune logical logs that only contains transactions older than 10 days from the current time, " +
             "or \"100k txs\" will keep the 100k latest transactions and prune any older transactions." )
-    public static final Setting<String> keep_logical_logs = buildSetting( "dbms.tx_log.rotation.retention_policy",
-            STRING, "7 days" ).constraint( illegalValueMessage( "must be `true`/`false` or " +
-                    "of format '<number><optional unit> <type>' for example `100M size` for " +
-                    "limiting logical log space on disk to 100Mb," +
-                    " or `200k txs` for limiting the number of transactions to keep to 200 000", matches(ANY) ) ).build();
+    @Dynamic
+    public static final Setting<String> keep_logical_logs =
+            buildSetting( "dbms.tx_log.rotation.retention_policy", STRING, "7 days" ).constraint( illegalValueMessage(
+                    "must be `true`, `false` or of format `<number><optional unit> <type>`. " +
+                            "Valid units are `k`, `M` and `G`. " +
+                            "Valid types are `files`, `size`, `txs`, `entries`, `hours` and `days`. " +
+                            "For example, `100M size` will limiting logical log space on disk to 100Mb," +
+                            " or `200k txs` will limiting the number of transactions to keep to 200 000", matches(
+                            "^(true|keep_all|false|keep_none|(\\d+[KkMmGg]?( (files|size|txs|entries|hours|days))))$" ) ) )
```

Comment:
i do not think we need to have this regexp here

Reply:
We do, otherwise, validation will not work for dynamic setting updates.

Reply:
ah, stupid me; missed \" quote and was reading it as part of descriptions string

Reply:
that strange feeling when you see a thumbs up emoji under `stupid me` comment... 🙉